packages/control-property-editor/src/App.tsx

```diff
@@ -27,6 +28,12 @@ export interface AppProps {
  */
 export default function App(appProps: AppProps): ReactElement {
     const { previewUrl } = appProps;
+    const urlWithoutHash = previewUrl.split('#')[0];
+    const queryStr = urlWithoutHash.split('?')[1];
+    const urlParams = new URLSearchParams(queryStr);
+    const scenario = urlParams.get('sap-ui-scenario') as Scenario;
+    const isAdpProject = scenario === 'ADAPTATION_PROJECT';
```

Comment:
Please use the `flexSettings` for reading the scenario https://github.com/SAP/open-ux-tools/blob/063ef51d1e2f0d906f4f295bc7776c32ab197a08/packages/preview-middleware-client/src/cpe/outline/index.ts#L17

Reply:
This is react app and we don't have access to rta.

Reply:
@GDamyanov There is a communication channel between the react app and the preview where RTA is running. Please read the scenario from the RTA and then send it to the react app.

Reply:
> @GDamyanov There is a communication channel between the react app and the preview where RTA is running. Please read the scenario from the RTA and then send it to the react app.

This way there is a huge delay for the post messaging between the iframe and react application (different loading times). This is really visible for the user. The most efficient way is to pass scenario in `previewUrl` to the `start` method, which is the entry point of the react application. This way when the react application is loading we already know the scenario and we do not have a huge delay.

Reply:
@nikmace Good point. If this is the case, then please add the `scenario` to the start options here https://github.com/SAP/open-ux-tools/blob/main/packages/control-property-editor/src/index.tsx#L28

Reply:
@nikmace @GDamyanov It is possible to read the `scenario` from the store. It is already there. Please see here https://github.com/SAP/open-ux-tools/blob/e9f1ea561af740d92b54f43bb13114be6b634d6e/packages/control-property-editor/src/index.tsx#L28

Reply:
> @nikmace @GDamyanov It is possible to read the `scenario` from the store. It is already there. Please see here
> 
> https://github.com/SAP/open-ux-tools/blob/a9de3d97f3678e8d54fe5525530a84e382b3e273/packages/control-property-editor/src/App.tsx#L42

Where is this scenario coming from? From post-messaging? If yes, that is exactly what I was talking about we will now be getting a big delay before we actually know in what scenario we are. In App.tsx you should instead pass the scenario to the URL and dispatch it in App.tsx. And then you can get it from the store by either using a useSelector or directly using the variable which you retrieved from previewUrl.

The result of the last commit with reverted changes will be is that first the react application will be loaded, second, the iframe will start being rendered (and we will see white container), then we receive a post message from it with the scenario and the redux store updates and we stop showing the iframe and show a dialog message.  The time between the react application load time and the iframe is noticeable not to say the least.

Reply:
@nikmace When I start the application the time between loading the react application and then the iframe is good. There is no significant delay.